src/main/java/seedu/address/model/moduleclass/UniqueModuleClassList.java

```diff
@@ -0,0 +1,156 @@
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateModuleClassException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the {@code ModuleClass} {@code target} in the list with {@code editedModuleClass}.
+     * {@code target} must exist in the list.
+     * The {@code ModuleClass} identity of {@code editedModuleClass} must not be the same
+     * as another existing {@code ModuleClass} in the list.
+     *
+     * @throws NullPointerException if the given argument is null.
+     * @throws ModuleClassNotFoundException if {@code target} does not exist in the list.
+     * @throws DuplicateModuleClassException if the identity of {@code editedModuleClass} already exists in the list.
+     */
+    public void setModuleClass(ModuleClass target, ModuleClass editedModuleClass)
+            throws NullPointerException, ModuleClassNotFoundException, DuplicateModuleClassException {
+        requireAllNonNull(target, editedModuleClass);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new ModuleClassNotFoundException();
+        }
+
+        if (!target.isSameModuleClass(editedModuleClass) && contains(editedModuleClass)) {
+            throw new DuplicateModuleClassException();
+        }
+
+        internalList.set(index, editedModuleClass);
+    }
+
+    public void setModuleClass(UniqueModuleClassList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code moduleClasses}.
+     * {@code moduleClasses} must not contain duplicate {@code ModuleClass}.
+     *
+     * @throws NullPointerException if any of the given arguments are null.
+     * @throws DuplicateModuleClassException if {@code moduleClasses} contains duplicate {@code ModuleClass}.
+     */
+    public void setModuleClass(List<ModuleClass> moduleClasses)
+            throws NullPointerException, DuplicateModuleClassException {
+        requireAllNonNull(moduleClasses);
+        if (!moduleClassesAreUnique(moduleClasses)) {
+            throw new DuplicateModuleClassException();
+        }
+
+        internalList.setAll(moduleClasses);
+    }
+
+    /**
+     * Removes the equivalent {@code ModuleClass} from the list.
+     * The {@code ModuleClass} must exist in the list.
+     *
+     * @throws NullPointerException if the given argument is null.
+     * @throws ModuleClassNotFoundException if {@code toRemove} does not exist in the list.
+     */
+    public void remove(ModuleClass toRemove) throws NullPointerException, ModuleClassNotFoundException {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new ModuleClassNotFoundException();
+        }
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
```

Comment:
Don't quite understand this. Specifically, don't understand what is a backing list and what is meant by unmodifiable

Reply:
This JavaDoc was directly adapted from a corresponding JavaDoc in the existing `UniqueStudentList` class as well. As such, I would be more inclined to retain the current expression since it is consistent with existing code.

The `backing list` refers to the `internalList` while unmodifiable means that any attempted modifications on the list would result in an `UnsupportedOperationException` being thrown.

Reply:
Ok lets retain it in that case